pkg/instrumentation/golang.go

```diff
@@ -36,9 +37,17 @@ func injectGoSDK(goSpec v1alpha1.Go, pod corev1.Pod) (corev1.Pod, error) {
 		return pod, fmt.Errorf("shared process namespace has been explicitly disabled")
 	}
 
-	containerNames, ok := pod.Annotations[annotationInjectContainerName]
-	if ok && len(strings.Split(containerNames, ",")) > 1 {
-		return pod, fmt.Errorf("go instrumentation cannot be injected into a pod using instrumentation.opentelemetry.io/container-names with more than 1 container")
+	// skip instrumentation when more than one containers provided
+	if featuregate.EnableMultiInstrumentationSupport.IsEnabled() {
+		containerNames, ok := pod.Annotations[annotationInjectGoContainersName]
+		if ok && len(strings.Split(containerNames, ",")) > 1 {
+			return pod, fmt.Errorf("go instrumentation cannot be injected into a pod using instrumentation.opentelemetry.io/go-container-names with more than 1 container")
+		}
+	} else {
+		containerNames, ok := pod.Annotations[annotationInjectContainerName]
+		if ok && len(strings.Split(containerNames, ",")) > 1 {
+			return pod, fmt.Errorf("go instrumentation cannot be injected into a pod using instrumentation.opentelemetry.io/container-names with more than 1 container")
+		}
```

Comment:
We can move `containerNames` and `ok` outside the condition, use the if-else conditions to set the values, and then do a single `if ok and len` check.